addons/web/static/tests/widgets/company_switcher_tests.js

```diff
@@ -125,6 +127,33 @@ QUnit.module('widgets', {
             });
         });
 
+        QUnit.test("Toggle multiple companies in one go in multiple company mode", async function (assert) {
+            /**
+             *          [x] **Company 1**          [x] **Company 1**
+             *  toggle->[ ] Company 2     ====>    [x] Company 2
+             *  toggle->[x] Company 3              [x] Company 3
+             */
+            assert.expect(2);
+            SwitchCompanyMenu.prototype.TOGGLE_DELAY = 1000;
+
+            const menu = await createSwitchCompanyMenu({
+                session: {
+                    ...this.session_mock_multi,
+                    setCompanies: function (mainCompanyId, companyIds) {
+                        assert.equal(mainCompanyId, 1, "Main company should not have changed");
+                        assert.equal(companyIds.length, 2, "First and Second company should be activated");
+                    },
+                },
+            });
+
+            await testUtils.dom.click(menu.$('.dropdown-toggle')); // open company switcher dropdown
+            await testUtils.dom.click(menu.$(`div[data-company-id="2"] div.toggle_company`));
+            await testUtils.dom.click(menu.$(`div[data-company-id="3"] div.toggle_company`));
+
+            await concurrency.delay(1000);
```

Comment:
to improve robustness, you could use a Promise here (`await prom;`), that you would have defined in the beginning of the test, and that you resolve in `setCompanies`:
```js
let resolver;
const prom = new Promise((r) => {
   resolver = r;
});
// ...
setCompanies: function (mainCompanyId, companyIds) {
    // ...
    resolver();
});
// ...
await prom;
menu.destroy();
```